src/main/java/org/sagebionetworks/bridge/exporter/BridgeExporter.java

```diff
@@ -0,0 +1,276 @@
+        // http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/java-dg-setup.html#set-up-creds for more
+        // info.
+        ddbClient = new DynamoDB(new AmazonDynamoDBClient());
+
+        // S3 client - move to Spring
+        AmazonS3Client s3Client = new AmazonS3Client();
+        S3Helper s3Helper = new S3Helper();
+        s3Helper.setS3Client(s3Client);
+
+        // synapse client
+        SynapseClient synapseClient = new SynapseClientImpl();
+        synapseClient.setUserName(config.getUsername());
+        synapseClient.setApiKey(config.getApiKey());
+
+        // because of a bug in the Java client, we need to properly log in to upload file handles
+        // see https://sagebionetworks.jira.com/browse/PLFM-3310
+        synapseClient.login(config.getUsername(), config.getPassword());
+
+        // synapse helper
+        SynapseHelper synapseHelper = new SynapseHelper();
+        synapseHelper.setBridgeExporterConfig(config);
+        synapseHelper.setS3Helper(s3Helper);
+        synapseHelper.setSynapseClient(synapseClient);
+
+        // schema Helper
+        schemaHelper = new UploadSchemaHelper();
+        schemaHelper.setDdbClient(ddbClient);
+        schemaHelper.init();
+
+        // export worker manager
+        manager = new ExportWorkerManager();
+        manager.setBridgeExporterConfig(config);
+        manager.setDdbClient(ddbClient);
+        manager.setS3Helper(s3Helper);
+        manager.setSchemaHelper(schemaHelper);
+        manager.setSynapseClient(synapseClient);
+        manager.setSynapseHelper(synapseHelper);
+        manager.setTodaysDateString(todaysDateString);
+        manager.init();
+    }
+
+    private void downloadHealthDataRecords() {
+        Stopwatch progressStopwatch = Stopwatch.createStarted();
+
+        // get key objects by querying uploadDate index
+        Table recordTable = ddbClient.getTable("prod-heroku-HealthDataRecord3");
+        Index recordTableUploadDateIndex = recordTable.getIndex("uploadDate-index");
+        Iterable<Item> recordKeyIter = recordTableUploadDateIndex.query("uploadDate", uploadDateString);
+
+        // re-query table to get values
+        for (Item oneRecordKey : recordKeyIter) {
+            // running count of records
+            int numTotal = incrementCounter("numTotal");
+            if (numTotal % PROGRESS_REPORT_PERIOD == 0) {
+                System.out.println("Num records so far: " + numTotal + " in "
+                        + progressStopwatch.elapsed(TimeUnit.SECONDS) + " seconds");
+            }
+            if (RECORD_LIMIT > 0 && numTotal > RECORD_LIMIT) {
+                break;
+            }
+
+            String recordId = oneRecordKey.getString("id");
+            try {
+                // re-query health data records to get values
+                Item oneRecord;
+                try {
+                    oneRecord = recordTable.getItem("id", recordId);
```

Comment:
This here will slow this down quite a bit and should be unnecessary. The query itself can return the values already